### DISCUSSION TITLE
The post request will not be submitted if the parameter is the default value

### DISCUSSION THREAD

**Discussion by jianminPei** [reactiveui/refit#1485] [Q&A] [answered]
if my request is suchs as："{
"productId ":123,
type:0
}" .
the post cant not transfer  type value. but I change it to " public int ? Type { get; set; }",the request is ok.

Thanks for helping me out.this is my code:
   
public class AsyncStateRequestDto
    {
        public string ProductId { get; set; }
        public int Type { get; set; }
    }

public interface ISMSProductApi : IJuaiApi
{
        [Post("/stateUpdate")]
        Task<AjaxResponse> StateUpdateAsync([Body] AsyncStateRequestDto requestDto);
}

**Comment by glennawatson** [comment, marked as answer]
This is a serializer setting, not a Refit bug. With `[Body]` the DTO is handed to your content serializer (System.Text.Json by default in current Refit).

`int Type` with value 0 is only omitted if your `JsonSerializerOptions` has `DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingDefault`. Making it `int?` works because then the property is non-default (it holds a boxed value), so it is no longer treated as "default" and gets written.

Fixes, pick one:

- Do not set `DefaultIgnoreCondition` to `WhenWritingDefault` (leave it at `Never`), so 0 is always written:

```csharp
var settings = new RefitSettings(
    new SystemTextJsonContentSerializer(new JsonSerializerOptions
    {
        DefaultIgnoreCondition = JsonIgnoreCondition.Never
    }));
```

- Or keep ignoring defaults globally but force this property to always serialize:

```csharp
[JsonIgnore(Condition = JsonIgnoreCondition.Never)]
public int Type { get; set; }
```

- Or keep it nullable (`int?`) if 0 vs "not sent" is a meaningful distinction in your API.

If you are on the Newtonsoft serializer instead, the equivalent knob is `NullValueHandling` / `DefaultValueHandling`.
